cmd/root.go

```diff
@@ -48,13 +48,23 @@ var (
 		Short: "Istio Manager",
 		PersistentPreRunE: func(cmd *cobra.Command, args []string) (err error) {
 			glog.V(2).Infof("Root flags: %#v", RootFlags)
+
+			if RootFlags.Kubeconfig == "" {
+				if os.Getenv("KUBECONFIG") != "" {
+					glog.V(2).Infof("Setting configuration from KUBECONFIG environment variable")
+					RootFlags.Kubeconfig = os.Getenv("KUBECONFIG")
+				}
+			}
+
 			Client, err = kube.NewClient(RootFlags.Kubeconfig, model.IstioConfig)
-			if err != nil {
+			if err != nil && RootFlags.Kubeconfig == "" {
```

Comment:
Is the second predicate necessary? Why not fallback to .kube/config even if KUBECONFIG is wrong?